src/platform/stm32f1/spi.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: MIT
+ * SPDX-FileCopyrightText: 2021 Rafael Silva <silvagracarafael@gmail.com>
+ */
+
+#pragma once
+
+#include <stm32f1xx.h>
+
+#include "platform/stm32f1/gpio.h"
+
+#define SPI_MSB_FIRST 0
+#define SPI_LSB_FIRST 1
+
+enum spi_mode {
+	SPI_MODE0 = 0,
+	SPI_MODE1 = 1,
+	SPI_MODE2 = 2,
+	SPI_MODE3 = 3,
+};
+
+enum spi_interface_no {
+	SPI_INTERFACE_1,
+#if defined(SPI2)
+	SPI_INTERFACE_2,
+#endif
+#if defined(SPI3)
+	SPI_INTERFACE_3,
+#endif
+};
+
+struct spi_drv_data_t {
```

Comment:
I feel like we should rename this `spi_interface_t` or something like that, making it clear this is independent of the HAL. Yes, this will be passed in as drv_data in the HAL but it may be used anywhere outside the HAL. In fact, the HAL just happens to use it. I wanted to make it clear it is not tied to the HAL. Does that make sense?

Reply:
I feel like leaking concepts to other parts of the code makes it harder to understand and identify stuff.

Reply:
sounds good to me